src/util.h

```diff
@@ -579,8 +579,10 @@ struct OnScopeLeaveImpl {
   }
   OnScopeLeaveImpl& operator=(OnScopeLeaveImpl&& other) {
     if (this == &other) return *this;
-    this->~OnScopeLeave();
```

Comment:
Actually, given that this is invalid code, because `~OnScopeLeave` is not even a function that exists, what do you think about deleting this operator overload instead of fixing it because it's completely unused?

Reply:
removed